### PR TITLE
Add square implementation

### PR DESCRIPTION
**Pull request by no-reply**
Cuts the long-side and offsets by 1/2 (rounding down) to center the image.

### CODE REVIEW

File: spec/lib/iiif/image_response/region_spec.rb [hunk 1]
```diff
@@ -112,4 +112,25 @@
...
+    end
+
+    it 'has square canonical value (width)' do
+      expect(described_class.new('square', 100, 201).canonical_value)
+        .to eq '0,50,100,100'
+    end
+
+    it 'has square canonical value (equal)' do
+      expect(described_class.new('square', 101, 101).canonical_value)
+        .to eq '0,0,101,101'
```
Comment (jpstroop) [inline spec/lib/iiif/image_response/region_spec.rb]: http://image-21.iiif.io/api/image/2.1/#canonical-uri-syntax in the table next to region:

> “full” if the whole image is requested, (including a “square” region of a square image)

so I'd `expect(described_class.new('square', 101, 101).canonical_value).to eq 'full'`

Comment (no-reply) [inline spec/lib/iiif/image_response/region_spec.rb]: :+1:

I had forgotten that this is where that discussion had landed. Will fix.

**Comment by jpstroop** [comment]
squash?

**Comment by no-reply** [comment]
done.

**Pull request merged**